index.js

```diff
@@ -1,19 +1,30 @@
 'use strict';
 
+function unhook(std, write) {
+	std.write = write;
+}
+
 function hook(type, opts, cb) {
 	if (typeof opts !== 'object') {
 		cb = opts;
 		opts = {};
 	}
 
-	opts = Object.assign({silent: true}, opts);
+	opts = Object.assign({
+		silent: true,
+		once: false
+	}, opts);
 
 	const std = process[type];
 	const write = std.write;
 
 	std.write = (str, enc, cb2) => {
 		const cbRet = cb(str, enc);
 
+		if (opts.once) {
+			unhook(std, write);
```

Comment:
Immediately, otherwise it's not really "once", as there might be more writes before the first write is done flushing.